airbyte-cdk/bulk/toolkits/load-low-code/src/main/kotlin/io/airbyte/cdk/load/discoverer/operation/InsertionMethod.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2025 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.load.discoverer.operation
+
+import com.fasterxml.jackson.databind.JsonNode
+import io.airbyte.cdk.load.command.ImportType
+import io.airbyte.cdk.load.data.AirbyteType
+import java.util.function.Predicate
+
+/**
+ * Describes the information related to how data is inserted in an object.
+ *
+ * Note that if matchingKeyPredicate is provided, it needs to return `true` to at least one of the
+ * properties. If this is not the case, the insertion method for this object will not be returned as
+ * part of the discover command. If not provided, it means that it is expected not to have matching
```

Comment:
If we do not have a valid case, should this be ignored or is it a hard fail situation?